apps/spruce/src/pages/adminSettings/Context.test.tsx

```diff
@@ -0,0 +1,76 @@
+import { act, renderHook, waitFor } from "@evg-ui/lib/test_utils";
+import { AdminSettingsTabRoutes } from "constants/routes";
+import { AdminSettingsProvider, useAdminSettingsContext } from "./Context";
+import { WritableAdminSettingsType } from "./tabs/types";
+
+describe("adminSettingsContext", () => {
+  it("ensure that tabs are initially saved", () => {
+    const { result } = renderHook(() => useAdminSettingsContext(), {
+      wrapper: AdminSettingsProvider,
+    });
+    expect(
+      result.current.getTab(AdminSettingsTabRoutes.Announcements).hasChanges,
+    ).toBe(false);
+  });
+  it("test check unsave settings", async () => {
+    const { result } = renderHook(() => useAdminSettingsContext(), {
+      wrapper: AdminSettingsProvider,
+    });
+
+    act(() => {
+      result.current.setInitialData({
+        [AdminSettingsTabRoutes.Announcements]: {
+          vars: [],
+        },
+      } as Record<WritableAdminSettingsType, any>);
+    });
+
+    act(() => {
+      result.current.updateForm(AdminSettingsTabRoutes.Announcements)({
+        formData: {
+          announcements: {
+            banner: "banner text!",
+          },
+        },
+        errors: [],
+      });
+    });
+
+    await waitFor(() => {
+      // console.log("After updateForm:", result.current.getChangedTabs());
+      expect(result.current.checkHasUnsavedChanges()).toEqual(true);
+    });
+  });
+
+  it("test get changed tabs", async () => {
+    const { result } = renderHook(() => useAdminSettingsContext(), {
+      wrapper: AdminSettingsProvider,
+    });
+
+    act(() => {
+      result.current.setInitialData({
+        [AdminSettingsTabRoutes.Announcements]: {
+          vars: [],
+        },
+      } as Record<WritableAdminSettingsType, any>);
```

Comment:
```suggestion
      } as Record<WritableAdminSettingsType, AnnouncementsFormState>);
```

Reply:
I had this but my text editor gives me red squiqqles so i avoided, but the test pass no problem 

Conversion of type '{ announcements: { announcements: { banner: string; BannerTheme: BannerTheme; }; }; }' to type 'Record<WritableAdminSettingsType, AnnouncementsFormState>' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
  Type '{ announcements: { announcements: { banner: string; BannerTheme: BannerTheme; }; }; }' is missing the following properties from type 'Record<WritableAdminSettingsType, AnnouncementsFormState>': "feature-flags", runners, web, authentication, and 4 more.ts(2352)
 
 
 the message if interested

Reply:
oh wait it does prevent the site from building : ( 

I was copying from here: 


https://github.com/evergreen-ci/ui/blob/7c1a35ca80cf2b1c53fa73594b2b2a10ded900c1/apps/spruce/src/pages/projectSettings/Context.test.tsx#L28